vault/resource_jwt_auth_backend.go

```diff
@@ -288,9 +312,19 @@ func jwtAuthBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	configuration := map[string]interface{}{}
 	for _, configOption := range matchingJwtMountConfigOptions {
-		// Set the configuration if the user has specified it, or the attribute is in the Diff
-		if _, ok := d.GetOkExists(configOption); ok || d.HasChange(configOption) {
-			configuration[configOption] = d.Get(configOption)
+		if configOption != "provider_config" {
```

Comment:
I feel like a switch statement here might make the flow a little bit clearer.

Reply:
I simplified this in general. Thanks!